### PR TITLE
[ENG-3949] Delete resource

### PR DESCRIPTION
**Pull request by adlius**
-   Ticket: [ENG-3949]
-   Feature flag: n/a

## Screenshot(s)
 
Screenshot from 2022-08-01 23-26-26
Screenshot from 2022-08-01 23-26-33

## QA Notes

[ENG-3949]: https://openscience.atlassian.net/browse/ENG-3949?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lib/osf-components/addon/components/resources-list/resource-card/template.hbs [hunk 1]
```diff
@@ -17,6 +17,19 @@
                 {{@resource.pid}}
             </OsfLink>
             {{#if @editable}}
+                <ResourcesList::DeleteResource
+                    @resource={{@resource}}
+                    @onDelete={{@onDelete}}
+                    as |modal|
+                >
+                    <Button
```
Comment (futa-ikeda) [inline lib/osf-components/addon/components/resources-list/resource-card/template.hbs:25]: This button needs an aria-label

**Review by futa-ikeda — CHANGES_REQUESTED** [review]
Should we use the `<DeleteButton>` component for this? If not, maybe we revisit the `<DeleteButton>` component after this project and turn it into a glimmer component?

File: lib/osf-components/addon/components/resources-list/delete-resource/component.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    @task
+    @waitFor
+    async deleteResource() {
+        try {
+            await this.args.resource.destroyRecord();
+            this.toast.info(this.intl.t('osf-components.resources-list.delete_resource.delete_success'));
+            this.args.onDelete();
+        } catch (e) {
+            this.toast.error(this.intl.t('osf-components.resources-list.delete_resource.delete_failure'));
```
Comment (brianjgeiger) [inline lib/osf-components/addon/components/resources-list/delete-resource/component.ts:26]: This should show the error from the API. `this.toast.error(getApiErrorMessage(e), errorMessage);`

**Review by brianjgeiger — CHANGES_REQUESTED** [review]
One additional thing that came up after a similar bug showed up elsewhere.

**Review by brianjgeiger — APPROVED** [review]

File: lib/osf-components/addon/components/resources-list/resource-card/template.hbs [hunk 3]
```diff
@@ -17,13 +17,26 @@
...
+                        <FaIcon @icon='trash-alt' @fixedWidth={{true}} />
+                    </Button>
+                </ResourcesList::DeleteResource>
                 <ResourcesList::EditResource
                     @resource={{@resource}}
                     as |modal|
                 >
                     <Button
                         local-class='EditButton'
-                        aria-label={{t 'osf-components.resources-list.edit_resource.edit_button_aria_label'}}
```
Comment (futa-ikeda) [inline lib/osf-components/addon/components/resources-list/resource-card/template.hbs:26]: Was this aria-label removed on purpose?

File: lib/osf-components/addon/components/resources-list/resource-card/component.ts [hunk 4]
```diff
@@ -11,6 +11,7 @@ interface ResourseCardArgs {
 
 export default class ResourceCardComponent extends Component<ResourseCardArgs> {
     @service intl!: Intl;
+    @service toast!: Toastr;
```
Comment (futa-ikeda) [inline lib/osf-components/addon/components/resources-list/resource-card/component.ts:14]: This seems unnecessary

**Review by futa-ikeda — CHANGES_REQUESTED** [review]
2 minor things

**Review by futa-ikeda — APPROVED** [review]

**Pull request merged**